### PR TITLE
Fe/main build

### PR DESCRIPTION
**Pull request by Eventyret**
Please see: [Notion Doc](https://www.notion.so/strapi/Internal-LMS-Documentation-0cc5605d3a974138ad9265b5de0d221e?pvs=4) for commands that are added to `package.json`

Let me know if you want any changes to it. 

- Created a top-level package.json with different scripts to start and stop development (this should simplify frontend work running everything we need)
- Created our Typescript copying script `copytypes.js`  To use the built-in `yarn strapi ts:generate-types` command. It will copy it into the frontend
- Configured TS types to include these new types.
- Configured tailwind to apply
- Configured ShadCN to the frontend.
- Moved to React Vanilla over Remix due to Remix been converted to react router

### CODE REVIEW

File: copytypes.js [hunk 1]
```diff
@@ -0,0 +1,80 @@
+const fs = require("fs").promises;
+const path = require("path");
+const { exec } = require("child_process");
+
+const destinationFolder = "frontend/types";
+
+// Adjust these paths as per your project structure
+const files = [
+  {
+    src: path.join(__dirname, "./backend/types/generated/contentTypes.d.ts"),
```
Comment (derrickmehaffy) [inline copytypes.js]: We should probably use an os agnostic package, while none of us use windows there might be some external users or fork/clone this.

**Review by derrickmehaffy — CHANGES_REQUESTED** [review]

File: copytypes.js (hunk 1 above)
Comment (Eventyret) [reply copytypes.js:10]: True I'll make it more agnostic but that also means other dependency for package.json

Comment (Eventyret) [reply copytypes.js:10]: Added agonistc way.

Comment (Eventyret) [reply copytypes.js]: Resolved made it agnostic.

**Comment by Eventyret** [comment]
Moved to React and Server components instead of, remix due to Remix been moved to react router.

**Review by derrickmehaffy — APPROVED** [review]

**Pull request merged**